modules/tinymce/src/core/main/ts/fmt/MergeFormats.ts

```diff
@@ -33,26 +35,28 @@ const mergeBackgroundColorAndFontSize = (dom: DOMUtils, format: ApplyFormat, var
   // nodes with font-size should have their own background color as well to fit the line-height (see TINY-882)
   if (format.styles && format.styles.backgroundColor) {
     processChildElements(node,
-      hasStyle(dom, 'fontSize'),
+      (elm) => hasStyle(dom, 'fontSize')(elm) && FormatUtils.isContentEditable(elm),
```

Comment:
I think I might lean more toward the latter mainly because the related `applyStyle` function still uses the currying.
I can do `const hasFontSize = hasStyle(dom, 'fontSize');` and use it that like you mentioned